### DISCUSSION TITLE
Using BO over larger bounds than x = [0,1]^d

### DISCUSSION THREAD

**Discussion by PraljakReps** [meta-pytorch/botorch#1249] [Q&A] [answered]
I have a question on using Expected Improvement acquisition with a standard Matern kernel GP model. I normalize the input data (latent vectors from a VAE) to` [0,1]^d`. However, during the `optimize_acqf() ` step, I increase the bounds to `[-3, 3]^d` so that the BO algorithm will sample new candidates outside the known latent space. For the BO protocol, I am using kriging believer;  at each queried candidate, I reintroduce the candidate into the training set with the GP predicted value. I am putting `q =1` too.  The main problem I am experiencing is that the BO algorithm is quite explorative when using EI acquisition when using larger bounds than the normalized training set. In particular, the sampled candidates are sampling the borders of the larger bounds ([-3,3]^d). 

Here is the training input data (colorbar is the y data).

![image](https://user-images.githubusercontent.com/61917512/172203668-5384d94e-b0b1-4a83-a766-7d02c2934631.png)

Here are the BO candidates:
![image](https://user-images.githubusercontent.com/61917512/172203713-aaac9805-d10c-44e7-9729-3fd09654224f.png)

Here are the BO candidates I aim to maximize `Norm r.e.`
![image](https://user-images.githubusercontent.com/61917512/172203840-b4457183-c37b-4f76-9317-75da87a764cb.png)

In terms of the last figure, the algorithm is sampling a lot of dead space (`norm r.e = 0`). 

How do I avoid the extreme exploration and avoid sampling obvious dead regions of the latent space while still increasing the bounds beyond the training set [0,1]^d?

**Comment by saitcakmak** [comment, marked as answer]
It is hard to say much from these plots only. If you share a reproducible example, e.g. a [colab](https://colab.research.google.com/) notebook, we can take a closer look. Here are things I'd check based on your description:
- You're optimizing EI in a much larger space than you have training data for. It is quite likely that the edges of the solution space will have large posterior variance, leading to large EI values. So, I wouldn't be surprised to see BO exploring the edges of the search space.
- What does the kriging believer / posterior mean look like at the edges of the solutions space? You're adding these to your model as training data, which will reduce the uncertainty near these points. However, if these predictions are "good" compared to the rest of your training data, they will only serve to encourage further sampling from these regions.
- Have you tried using `q>1` instead of conditioning on the posterior mean? If your `q` is too large for joint optimization, you can also pass `sequential=True`.
- What's the dimension of the input space? The above issues will only get worse as `d` increases.
